snapshot/lib/snapshot/test_command_generator.rb

```diff
@@ -78,9 +78,17 @@ def find_device(device_name, os_version = Snapshot.config[:ios_version])
         end
       end
 
-      def device_udid(device_name, os_version = Snapshot.config[:ios_version])
+      def device_udid(device_name)
+        os = device_name =~ /^Apple TV/ ? "tvOS" : "iOS"
+        os_version = Snapshot.config[:ios_version] || Snapshot::LatestOsVersion.version(os)
+
         device = find_device(device_name, os_version)
 
+        if device.nil?
+          UI.user_error!("No device found named '#{device_name}' for version '#{os_version}'")
```

Comment:
This will raise an exception and stop running _snapshot_ I think. Is that the behavior we want? Can you reproduce the error so that it shows the error message and share the output with us?